### DISCUSSION TITLE
iOS 4.2.110 版本崩溃问题

### DISCUSSION THREAD

**Discussion by htxs** [Tencent/libpag#2154] [General]
libpag 版本信息: 4.2.110  （pod 'libpag', :git => 'https://github.com/libpag/libpag-ios.git', :tag => '4.2.110'）
系统版本分布情况 : iOS 15、iOS 16、iOS 17 均存在，无明显的系统版本特征
是否存在复现路径: 线上崩溃，未复现
堆栈详情如下:

Last Exception :
0  AGXMetalA9                     0x00000001dd273a90 0x00000001dd197000 + 903824
1  AGXMetalA9                     0x00000001dd2551ec 0x00000001dd197000 + 778732
2  AGXMetalA9                     0x00000001dd2551ec 0x00000001dd197000 + 778732
3  AGXMetalA9                     0x00000001dd253bf0 0x00000001dd197000 + 773104
4  AppleMetalGLRenderer           0x00000001dd2f4c70 gldModifyTexSubImage + 2132
5  GLEngine _glTexSubImage2D_Exec + 1716
6  OpenGLES _glTexSubImage2D + 80
7  libpag tgfx::GLGpu::writePixels(tgfx::TextureSampler const*, tgfx::Rect, void const*, unsigned long) (GLGpu.cpp:103)
8  libpag tgfx::SubmitYUVTexture(tgfx::Context*, tgfx::YUVData const*, std::__1::vector<std::__1::unique_ptr<tgfx::TextureSampler, std::__1::default_delete<tgfx::TextureSampler> >, std::__1::allocator<std::__1::unique_ptr<tgfx::TextureSampler, std::__1::default_delete<tgfx::TextureSampler> > > > const*) + 220
9  libpag tgfx::Texture::MakeI420(tgfx::Context*, tgfx::YUVData const*, tgfx::YUVColorSpace) + 632
10 libpag tgfx::ImageGeneratorTextureProxy::onMakeTexture(tgfx::Context*) (shared_ptr.h:701)
11 libpag tgfx::TextureProxy::instantiate() (TextureProxy.cpp:57)
12 libpag tgfx::DrawingManager::flush(tgfx::Semaphore*) (DrawingManager.cpp:60)
13 libpag                         tgfx::Surface::flush(tgfx::BackendSemaphore*) (Surface.cpp:159)
14 libpag                         pag::PAGSurface::draw(pag::RenderCache*, std::__1::shared_ptr<pag::Graphic>, pag::BackendSemaphore*, bool) (PAGSurface.cpp:147)
15 libpag pag::PAGPlayer::flushInternal(pag::BackendSemaphore*) (PAGPlayer.cpp:294)
16 libpag pag::PAGPlayer::flush() (PAGPlayer.cpp:284)
17 libpag -[PAGView flush] (PAGView.mm:441)
18 libpag                         _26-[PAGView updateViewAsync]_block_invoke (PAGView.mm:487)
19 Foundation                     _NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__ + 16
20 Foundation                     -[NSBlockOperation main] + 100
21 Foundation                     _NSOPERATION_IS_INVOKING_MAIN__ + 20
22 Foundation                     -[NSOperation start] + 764
23 Foundation                     _NSOPERATIONQUEUE_IS_STARTING_AN_OPERATION__ + 20
24 Foundation                     _NSOQSchedule_f + 180
25 libdispatch.dylib _dispatch_block_async_invoke2 + 104
26 libdispatch.dylib _dispatch_client_callout + 16
27 libdispatch.dylib _dispatch_continuation_pop$VARIANT$mp + 440
28 libdispatch.dylib              _dispatch_async_redirect_invoke + 588
29 libdispatch.dylib _dispatch_root_queue_drain + 340
30 libdispatch.dylib _dispatch_worker_thread2 + 172
31 libsystem_pthread.dylib _pthread_wqthread + 224

Exception Type: SIGSEGV SEGV_ACCERR
Exception Codes: fault addr: 0x0000000148810000
Crashed Thread: 54

**Comment by kevingpqi123** [comment]
可以升级至 4.3.57看下，之前收到过类似的反馈，可以让你们的测试同学在相应场景复现下，如果有稳定的复现路径，可以加速问题的解决

**Comment by CodeJhF** [comment]
这条我们在当时的回复里已经给出结论，这里就不重复贴了，先关闭本条。
如果按当时的结论排查后仍有问题，欢迎回复补充最小复现 demo 与版本号（4.4 / 4.5），我们会重新看。（stale-close）
